File: test/fsharpTests/CharTest.fs
open Tablecloth
open Expecto

[<Tests>]
let tests =
  testList
  "Char"
[testCase "fromCode(97)" 
<| fun _ -> 
    let expected = Some('a')
    Expect.equal expected (Char.fromCode 97) "error"
testCase "fromCode(-1)" 
<| fun _ -> 
    let expected = None
    Expect.equal expected (Char.fromCode -1) "error"
testCase "fromCode(256)" 
<| fun _ -> 
    let expected = None
    Expect.equal expected (Char.fromCode 256) "error"
testCase "fromString("a")" 
<| fun _ -> 
    let expected = Some('a')
    Expect.equal expected (Char.fromString "a") "error"
testCase "fromString("abc")" 
<| fun _ -> 
    let expected = None
    Expect.equal expected (Char.fromString "abc") "error"
testCase "fromString("")" 
<| fun _ -> 
    let expected = None
    Expect.equal expected (Char.fromString "") "error"
testCase "isAlphanumeric('A')" 
<| fun _ -> 
    let expected = true
    Expect.equal expected (Char.isAlphanumeric 'A') "error"
testCase "isAlphanumeric('?')" 
<| fun _ -> 
    let expected = false
    Expect.equal expected (Char.isAlphanumeric '?') "error"
testCase "isDigit('0')" 
<| fun _ -> 
    let expected = true
    Expect.equal expected (Char.isDigit '0') "error"
testCase "isDigit('1')" 
<| fun _ -> 
    let expected = true
    Expect.equal expected (Char.isDigit '1') "error"
testCase "isDigit('2')" 
<| fun _ -> 
    let expected = true
    Expect.equal expected (Char.isDigit '2') "error"
testCase "isDigit('3')" 
<| fun _ -> 
    let expected = true
    Expect.equal expected (Char.isDigit '3') "error"
testCase "isDigit('4')" 
<| fun _ -> 
    let expected = true
    Expect.equal expected (Char.isDigit '4') "error"
testCase "isDigit('5')" 
<| fun _ -> 
    let expected = true
    Expect.equal expected (Char.isDigit '5') "error"
testCase "isDigit('6')" 
<| fun _ -> 
    let expected = true
    Expect.equal expected (Char.isDigit '6') "error"
testCase "isDigit('7')" 
<| fun _ -> 
    let expected = true
    Expect.equal expected (Char.isDigit '7') "error"
testCase "isDigit('8')" 
<| fun _ -> 
    let expected = true
    Expect.equal expected (Char.isDigit '8') "error"
testCase "isDigit('9')" 
<| fun _ -> 
    let expected = true
    Expect.equal expected (Char.isDigit '9') "error"
testCase "isDigit('a')" 
<| fun _ -> 
    let expected = false
    Expect.equal expected (Char.isDigit 'a') "error"
testCase "isLetter('A')" 
<| fun _ -> 
    let expected = true
    Expect.equal expected (Char.isLetter 'A') "error"
testCase "isLetter('7')" 
<| fun _ -> 
    let expected = false
    Expect.equal expected (Char.isLetter '7') "error"
testCase "isLetter(' ')" 
<| fun _ -> 
    let expected = false
    Expect.equal expected (Char.isLetter ' ') "error"
testCase "isLetter('\n')" 
<| fun _ -> 
    let expected = false
    Expect.equal expected (Char.isLetter '\n') "error"
testCase "isLetter('\001')" 
<| fun _ -> 
    let expected = false
    Expect.equal expected (Char.isLetter '\001') "error"
testCase "isLetter('\236')" 
<| fun _ -> 
    let expected = false
    Expect.equal expected (Char.isLetter '\236') "error"
testCase "isLowercase('a')" 
<| fun _ -> 
    let expected = true
    Expect.equal expected (Char.isLowercase 'a') "error"
testCase "isLowercase('7')" 
<| fun _ -> 
    let expected = false
    Expect.equal expected (Char.isLowercase '7') "error"
testCase "isLowercase('\236')" 
<| fun _ -> 
    let expected = false
    Expect.equal expected (Char.isLowercase '\236') "error"
testCase "isPrintable('~')" 
<| fun _ -> 
    let expected = true
    Expect.equal expected (Char.isPrintable '~') "error"
testCase "isPrintable(fromCode(31) |> Option.map(~f=isPrintable))" 
<| fun _ -> 
    let expected = Some(false)
    Expect.equal expected (Char.isPrintable fromCode(31) |> Option.map(~f=isPrintable)) "error"
testCase "isUppercase('A')" 
<| fun _ -> 
    let expected = true
    Expect.equal expected (Char.isUppercase 'A') "error"
testCase "isUppercase('7')" 
<| fun _ -> 
    let expected = false
    Expect.equal expected (Char.isUppercase '7') "error"
testCase "isUppercase('\237')" 
<| fun _ -> 
    let expected = false
    Expect.equal expected (Char.isUppercase '\237') "error"
testCase "isWhitespace(' ')" 
<| fun _ -> 
    let expected = true
    Expect.equal expected (Char.isWhitespace ' ') "error"
testCase "isWhitespace('a')" 
<| fun _ -> 
    let expected = false
    Expect.equal expected (Char.isWhitespace 'a') "error"
testCase "toCode('a')" 
<| fun _ -> 
    let expected = 97
    Expect.equal expected (Char.toCode 'a') "error"
testCase "toDigit('0')" 
<| fun _ -> 
    let expected = Some(0)
    Expect.equal expected (Char.toDigit '0') "error"
testCase "toDigit('8')" 
<| fun _ -> 
    let expected = Some(8)
    Expect.equal expected (Char.toDigit '8') "error"
testCase "toDigit('a')" 
<| fun _ -> 
    let expected = None
    Expect.equal expected (Char.toDigit 'a') "error"
testCase "toLowercase('A')" 
<| fun _ -> 
    let expected = 'a'
    Expect.equal expected (Char.toLowercase 'A') "error"
testCase "toLowercase('a')" 
<| fun _ -> 
    let expected = 'a'
    Expect.equal expected (Char.toLowercase 'a') "error"
testCase "toLowercase('7')" 
<| fun _ -> 
    let expected = '7'
    Expect.equal expected (Char.toLowercase '7') "error"
testCase "toLowercase('\233')" 
<| fun _ -> 
    let expected = '\233'
    Expect.equal expected (Char.toLowercase '\233') "error"
testCase "toString('a')" 
<| fun _ -> 
    let expected = "a"
    Expect.equal expected (Char.toString 'a') "error"
testCase "toUppercase('a')" 
<| fun _ -> 
    let expected = 'A'
    Expect.equal expected (Char.toUppercase 'a') "error"
testCase "toUppercase('A')" 
<| fun _ -> 
    let expected = 'A'
    Expect.equal expected (Char.toUppercase 'A') "error"
testCase "toUppercase('7')" 
<| fun _ -> 
    let expected = '7'
    Expect.equal expected (Char.toUppercase '7') "error"
testCase "toUppercase('\233')" 
<| fun _ -> 
    let expected = '\233'
    Expect.equal expected (Char.toUppercase '\233') "error"
]

File: test/ocamlTests/CharTest.ml
open Tablecloth
open AlcoJest

let suite =
 suite "Char" (fun () ->
   let open Char in
test "fromCode(97)" (fun () -> expect (Char.fromCode 97) |> toEqual (let open Eq in option char) Some('a')) ; 
test "fromCode(-1)" (fun () -> expect (Char.fromCode -1) |> toEqual (let open Eq in option char) None) ; 
test "fromCode(256)" (fun () -> expect (Char.fromCode 256) |> toEqual (let open Eq in option char) None) ; 
test "fromString("a")" (fun () -> expect (Char.fromString "a") |> toEqual (let open Eq in option char) Some('a')) ; 
test "fromString("abc")" (fun () -> expect (Char.fromString "abc") |> toEqual (let open Eq in option char) None) ; 
test "fromString("")" (fun () -> expect (Char.fromString "") |> toEqual (let open Eq in option char) None) ; 
test "isAlphanumeric('A')" (fun () -> expect (Char.isAlphanumeric 'A') |> toEqual Eq.bool true) ; 
test "isAlphanumeric('?')" (fun () -> expect (Char.isAlphanumeric '?') |> toEqual Eq.bool false) ; 
test "isDigit('0')" (fun () -> expect (Char.isDigit '0') |> toEqual Eq.bool true) ; 
test "isDigit('1')" (fun () -> expect (Char.isDigit '1') |> toEqual Eq.bool true) ; 
test "isDigit('2')" (fun () -> expect (Char.isDigit '2') |> toEqual Eq.bool true) ; 
test "isDigit('3')" (fun () -> expect (Char.isDigit '3') |> toEqual Eq.bool true) ; 
test "isDigit('4')" (fun () -> expect (Char.isDigit '4') |> toEqual Eq.bool true) ; 
test "isDigit('5')" (fun () -> expect (Char.isDigit '5') |> toEqual Eq.bool true) ; 
test "isDigit('6')" (fun () -> expect (Char.isDigit '6') |> toEqual Eq.bool true) ; 
test "isDigit('7')" (fun () -> expect (Char.isDigit '7') |> toEqual Eq.bool true) ; 
test "isDigit('8')" (fun () -> expect (Char.isDigit '8') |> toEqual Eq.bool true) ; 
test "isDigit('9')" (fun () -> expect (Char.isDigit '9') |> toEqual Eq.bool true) ; 
test "isDigit('a')" (fun () -> expect (Char.isDigit 'a') |> toEqual Eq.bool false) ; 
test "isLetter('A')" (fun () -> expect (Char.isLetter 'A') |> toEqual Eq.bool true) ; 
test "isLetter('7')" (fun () -> expect (Char.isLetter '7') |> toEqual Eq.bool false) ; 
test "isLetter(' ')" (fun () -> expect (Char.isLetter ' ') |> toEqual Eq.bool false) ; 
test "isLetter('\n')" (fun () -> expect (Char.isLetter '\n') |> toEqual Eq.bool false) ; 
test "isLetter('\001')" (fun () -> expect (Char.isLetter '\001') |> toEqual Eq.bool false) ; 
test "isLetter('\236')" (fun () -> expect (Char.isLetter '\236') |> toEqual Eq.bool false) ; 
test "isLowercase('a')" (fun () -> expect (Char.isLowercase 'a') |> toEqual Eq.bool true) ; 
test "isLowercase('7')" (fun () -> expect (Char.isLowercase '7') |> toEqual Eq.bool false) ; 
test "isLowercase('\236')" (fun () -> expect (Char.isLowercase '\236') |> toEqual Eq.bool false) ; 
test "isPrintable('~')" (fun () -> expect (Char.isPrintable '~') |> toEqual Eq.bool true) ; 
test "isUppercase('A')" (fun () -> expect (Char.isUppercase 'A') |> toEqual Eq.bool true) ; 
test "isUppercase('7')" (fun () -> expect (Char.isUppercase '7') |> toEqual Eq.bool false) ; 
test "isUppercase('\237')" (fun () -> expect (Char.isUppercase '\237') |> toEqual Eq.bool false) ; 
test "isWhitespace(' ')" (fun () -> expect (Char.isWhitespace ' ') |> toEqual Eq.bool true) ; 
test "isWhitespace('a')" (fun () -> expect (Char.isWhitespace 'a') |> toEqual Eq.bool false) ; 
test "toCode('a')" (fun () -> expect (Char.toCode 'a') |> toEqual Eq.int 97) ; 
test "toDigit('0')" (fun () -> expect (Char.toDigit '0') |> toEqual (let open Eq in option int) Some(0)) ; 
test "toDigit('8')" (fun () -> expect (Char.toDigit '8') |> toEqual (let open Eq in option int) Some(8)) ; 
test "toDigit('a')" (fun () -> expect (Char.toDigit 'a') |> toEqual (let open Eq in option int) None) ; 
test "toLowercase('A')" (fun () -> expect (Char.toLowercase 'A') |> toEqual Eq.char 'a') ; 
test "toLowercase('a')" (fun () -> expect (Char.toLowercase 'a') |> toEqual Eq.char 'a') ; 
test "toLowercase('7')" (fun () -> expect (Char.toLowercase '7') |> toEqual Eq.char '7') ; 
test "toLowercase('\233')" (fun () -> expect (Char.toLowercase '\233') |> toEqual Eq.char '\233') ; 
test "toString('a')" (fun () -> expect (Char.toString 'a') |> toEqual Eq.string "a") ; 
test "toUppercase('a')" (fun () -> expect (Char.toUppercase 'a') |> toEqual Eq.char 'A') ; 
test "toUppercase('A')" (fun () -> expect (Char.toUppercase 'A') |> toEqual Eq.char 'A') ; 
test "toUppercase('7')" (fun () -> expect (Char.toUppercase '7') |> toEqual Eq.char '7') ; 
test "toUppercase('\233')" (fun () -> expect (Char.toUppercase '\233') |> toEqual Eq.char '\233') ; 
)